src/main/kotlin/fi/metatavu/lipsanen/projects/milestones/tasks/comments/TaskCommentsApiImpl.kt

```diff
@@ -0,0 +1,174 @@
+                return@async createBadRequest("Referenced user $referencedUserId is not an assignee of the task")
+            }
+        }
+        val createdComment = taskCommentController.createTaskComment(task, taskComment, userId)
+        createOk(taskCommentTranslator.translate(createdComment))
+    }.asUni()
+
+    @RolesAllowed(UserRole.ADMIN.NAME, UserRole.USER.NAME)
+    override fun findTaskComment(projectId: UUID, milestoneId: UUID, taskId: UUID, commentId: UUID): Uni<Response> =
+        CoroutineScope(vertx.dispatcher()).async {
+            val userId = loggedUserId ?: return@async createUnauthorized(UNAUTHORIZED)
+
+            val (projectMilestone, errorResponse) = getProjectMilestoneAccessRights(projectId, milestoneId, userId)
+            if (errorResponse != null) return@async errorResponse
+
+            val task = taskController.find(projectMilestone!!.first, taskId) ?: return@async createNotFound(
+                createNotFoundMessage(TASK, taskId)
+            )
+
+            val comment = taskCommentController.find(commentId)
+            if (comment == null || comment.task.id != task.id) return@async createNotFound(
+                createNotFoundMessage(TASK_COMMENT, commentId)
+            )
+
+            createOk(taskCommentTranslator.translate(comment))
+        }.asUni()
+
+    @WithTransaction
+    @RolesAllowed(UserRole.ADMIN.NAME, UserRole.USER.NAME)
+    override fun updateTaskComment(
+        projectId: UUID,
+        milestoneId: UUID,
+        taskId: UUID,
+        commentId: UUID,
+        taskComment: TaskComment
+    ): Uni<Response> = CoroutineScope(vertx.dispatcher()).async {
+        val userId = loggedUserId ?: return@async createUnauthorized(UNAUTHORIZED)
+        if (taskComment.taskId != taskId) return@async createBadRequest("Task id in path and body do not match")
+
+        val (projectMilestone, errorResponse) = getProjectMilestoneAccessRights(projectId, milestoneId, userId)
+        if (errorResponse != null) return@async errorResponse
+
+        val task = taskController.find(projectMilestone!!.first, taskId) ?: return@async createNotFound(
+            createNotFoundMessage(TASK, taskId)
+        )
+
+        val comment = taskCommentController.find(commentId) ?: return@async createNotFound(
+            createNotFoundMessage(TASK_COMMENT, commentId)
+        )
+
+        if (comment.task.id != task.id) return@async createNotFound(
+            createNotFoundMessage(TASK_COMMENT, commentId)
+        )
+
+        if (comment.creatorId != userId) return@async createForbidden("Cannot edit not own comment")
+
+        taskComment.referencedUsers.forEach { referencedUserId ->
+            if (!projectController.hasAccessToProject(task.milestone.project, referencedUserId)) {
+                return@async createBadRequest("Referenced user $referencedUserId is not an assignee of the task")
+            }
+        }
```

Comment:
Same here, I would consider a case when User would want to reference a non-assignee as a possible one. I.e. task assignee (person doing the task) wants to reference their accountant / project manager in a comment. Those people are likely not assigned to this same task

Reply:
Fixed the message